packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-checkbox-group/gux-form-field-checkbox-group.service.ts

```diff
@@ -0,0 +1,52 @@
+import simulateNativeEvent from '@utils/dom/simulate-native-event';
+
+export function setAllCheckboxInputs(
+  root: HTMLElement,
+  checked: boolean
+): void {
+  getCheckboxInputs(root).forEach(checkboxInput => {
+    if (checkboxInput.checked !== checked) {
+      checkboxInput.checked = checked;
+      simulateNativeEvent(checkboxInput, 'input');
+      simulateNativeEvent(checkboxInput, 'change');
+    }
+  });
+}
+
+export function setMainCheckboxElementCheckedState(
+  root: HTMLElement,
+  mainCheckboxElement: HTMLInputElement
+): void {
+  if (mainCheckboxElement) {
+    const { count, total } = getSelectedColumnCount(root);
+    if (count === 0) {
+      mainCheckboxElement.indeterminate = false;
+      mainCheckboxElement.checked = false;
+    } else if (count === total) {
+      mainCheckboxElement.indeterminate = false;
+      mainCheckboxElement.checked = true;
+    } else {
+      mainCheckboxElement.indeterminate = true;
+    }
+  }
+}
+
+export function getSelectedColumnCount(root: HTMLElement): {
+  count: number;
+  total: number;
+} {
+  const total = getCheckboxInputs(root).length;
+  const count = getCheckedCheckboxInputs(root).length;
```

Comment:
Thoughts on adding more specificity to the variable names `total` and `count`? If I look closer at this code I can see `count` refers to the checked checkboxes and `total` is the number of checkbox inputs but calling this method it might not be immediately intuitive.